### PR TITLE
feat(legend): support customizing marker icon of legend item

### PR DESCRIPTION
**Pull request by eraop**
## Description

> Support custom point style configuration.

## Screenshots

- [x] This PR does not have any UI modifications  
image
image

image

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation or example update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and example

### CODE REVIEW

File: website/docs/tutorials/demo-charts/line.mdx [hunk 1]
```diff
@@ -91,6 +91,7 @@ function showComponent() {
     },
   ];
   const options = {
+    legend: { itemSelectable: false },
```
Comment (bndynet) [inline website/docs/tutorials/demo-charts/line.mdx:94]: The `itemSelectable` will be renamed to `selectable`.

File: src/types/chart.legend.types.ts [hunk 2]
```diff
@@ -12,5 +12,6 @@ export interface LegendOptions {
   onItemClick?(context: LegendItemClickContext): void;
   display?: boolean;
   position?: Position;
+  pointStyle?: PointStyle;
```
Comment (bndynet) [inline src/types/chart.legend.types.ts:15]: Change to `markerStyle`.

File: website/docs/tutorials/demo-charts/line.mdx (hunk 1 above)
Comment (eraop) [reply website/docs/tutorials/demo-charts/line.mdx]: fixed

File: src/types/chart.legend.types.ts (hunk 2 above)
Comment (eraop) [reply src/types/chart.legend.types.ts]: fixed

**Review by bndynet — APPROVED** [review]

**Pull request merged**